<launch>

  <arg name="custom_config" default="" />

    <!-- other args -->
  <arg name="standalone" default="true" />
  <arg name="debug" default="false" />

  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="load" />
  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="$(arg UAV_NAME)_nodelet_manager" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <arg name="obstacle_file" default="$(find task_03_gazebo_resources)/models/column/meshes/column.dae" />

  <node name="plotter" pkg="nodelet" type="nodelet" args="$(arg nodelet) task_03_wrapper/Plotter $(arg nodelet_manager)" output="screen" launch-prefix="$(arg launch_prefix)">

    <rosparam file="$(find task_03_wrapper)/config/plotter.yaml" />

      <!-- Load a user param file -->
    <rosparam if="$(eval not arg('custom_config') == '')" file="$(arg custom_config)" />

    <param name="obstacle_file" value="$(arg obstacle_file)" />

      <!-- subscribers -->
    <remap from="~target_pose_in" to="/target_pose" />

      <!-- publishers -->
    <remap from="~rviz_markers_out" to="~rviz_markers" />
    <remap from="~true_target_out" to="~true_target" />
    <remap from="~true_target_path_out" to="~true_target_path" />

  </node>

</launch>
